Monytor.WebApi/Startup.cs

```diff
@@ -22,19 +21,28 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            var url = Configuration["database:url"];
-            var databaseName = Configuration["database:name"];
-            var store = RavenHelper.CreateStore(url, databaseName);
-
-            services.AddSingleton<IDocumentStore>(store);
-            services.AddTransient<ICollectorService, CollectorService>();
-            services.AddTransient<ISeriesRepository, SeriesRepository>();
-            services.AddTransient<IViewCollectionService, ViewCollectionService>();
-            services.AddTransient<IDashboardRepository, DashboardRepository>();
+            SetupDatabase(services, Configuration);
+            services.AddScoped<ICollectorService, CollectorService>();
+            services.AddScoped<IViewCollectionService, ViewCollectionService>();
 
             services.AddCors();
             services.AddMvc();            
         }
+        private static void SetupDatabase(IServiceCollection services, IConfiguration appConfig) {
```

Comment:
Same code already exists in Monytor.Startup/Bootstrapper.cs. If possible try to eliminate duplicate code. Maybe use "central" factory methode.

Reply:
Neary yeah, but the DI is bound the the specific container, so can't refactor much.